### PR TITLE
[ST] Modify CustomResourceStatus STs

### PR DESCRIPTION
**Pull request by jankalinic**
### Type of change

- Refactoring

### Description

This PR modifies the CustomResourceStatusST class as there are some tests cases covered in other test classes and some test just need different checks to better cover strimzi needs.

### Checklist

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally

### CODE REVIEW

File: systemtest/src/test/java/io/strimzi/systemtest/operators/CustomResourceStatusST.java [hunk 1]
```diff
@@ -162,72 +164,40 @@ void testKafkaUserStatus(ExtensionContext extensionContext) {
...
-        LOGGER.info("Checking status of deployed KafkaUser");
-        Condition kafkaCondition = KafkaUserResource.kafkaUserClient().inNamespace(Constants.TEST_SUITE_NAMESPACE).withName(userName).get().getStatus().getConditions().get(0);
-        LOGGER.info("KafkaUser Status: {}", kafkaCondition.getStatus());
-        LOGGER.info("KafkaUser Type: {}", kafkaCondition.getType());
-        assertThat("KafkaUser is in wrong state!", kafkaCondition.getType(), is(Ready.toString()));
-        LOGGER.info("KafkaUser is in desired state: Ready");
-    }
-
-    @ParallelTest
-    void testKafkaUserStatusNotReady(ExtensionContext extensionContext) {
```
Comment (henryZrncik) [inline systemtest/src/test/java/io/strimzi/systemtest/operators/CustomResourceStatusST.java:176]: maybe this PR could be also used to document tests that are modified by these changes.

**Review by henryZrncik — APPROVED** [review]
LGTM.

File: systemtest/src/test/java/io/strimzi/systemtest/operators/CustomResourceStatusST.java (hunk 1 above)
Comment (jankalinic) [reply systemtest/src/test/java/io/strimzi/systemtest/operators/CustomResourceStatusST.java:176]: This is very good point, i will do this in separate PR for all the test cases. I think we can finally start to do that in other test suites as well.

Comment (henryZrncik) [reply systemtest/src/test/java/io/strimzi/systemtest/operators/CustomResourceStatusST.java:176]: works for me 👍

**Review by see-quick — APPROVED** [review]
👍

**Review by Frawless — APPROVED** [review]

**Comment by jankalinic** [comment]
@strimzi-ci run tests --profile=regression --install-type=bundle --cluster-type=ocp --cluster-version=4.14 --testcase=CustomResourceStatusST

**Pull request merged**